beacon-chain/rpc/prysm/v1alpha1/validator/blocks.go

```diff
@@ -46,6 +47,53 @@ func (vs *Server) StreamBlocksAltair(req *ethpb.StreamBlocksRequest, stream ethp
 	}
 }
 
+// StreamSlots sends a block's slot to clients every single time a block is received by the beacon node.
+func (vs *Server) StreamSlots(req *ethpb.StreamSlotsRequest, stream ethpb.BeaconNodeValidator_StreamSlotsServer) error {
+	ch := make(chan *feed.Event, 1)
+	var sub event.Subscription
+	if req.VerifiedOnly {
+		sub = vs.StateNotifier.StateFeed().Subscribe(ch)
+	} else {
+		sub = vs.BlockNotifier.BlockFeed().Subscribe(ch)
+	}
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case ev := <-ch:
+			var s primitives.Slot
+			if req.VerifiedOnly {
+				if ev.Type != statefeed.BlockProcessed {
+					continue
+				}
+				data, ok := ev.Data.(*statefeed.BlockProcessedData)
+				if !ok || data == nil {
+					continue
+				}
+				s = data.Slot
+			} else {
+				if ev.Type != blockfeed.ReceivedBlock {
+					continue
+				}
+				data, ok := ev.Data.(*blockfeed.ReceivedBlockData)
+				if !ok || data == nil {
+					continue
+				}
+				s = data.SignedBlock.Block().Slot()
+			}
```

Comment:
Is there a use case for non verified slot received? is this just for beacon api?

Reply:
no, this is just legacy code, I didn't feel like removing the field to keep it as close as it was before, but we can remove it if needed.